core/store/orm/orm.go

```diff
@@ -614,29 +614,33 @@ func (orm *ORM) AnyJobWithType(taskTypeName string) (bool, error) {
 	return found, ignoreRecordNotFound(rval)
 }
 
-// CreateTx returns a transaction by its surrogate key, if it exists, or
+// CreateTx finds and overwrites a transaction by its surrogate key, if it exists, or
 // creates it
 func (orm *ORM) CreateTx(tx *models.Tx) (*models.Tx, error) {
 	orm.MustEnsureAdvisoryLock()
 
 	err := orm.convenientTransaction(func(dbtx *gorm.DB) error {
 		var query *gorm.DB
+		foundTx := models.Tx{}
 		if tx.SurrogateID.Valid {
-			query = dbtx.First(&models.Tx{}, "surrogate_id = ?", tx.SurrogateID.ValueOrZero())
+			query = dbtx.First(&foundTx, "surrogate_id = ?", tx.SurrogateID.ValueOrZero())
+			// TODO: Put ON CONFLICT REPLACE ALL here
+			// https://github.com/sljeff/gorm/pull/5
 		} else {
-			query = dbtx.First(&models.Tx{}, "hash = ?", tx.Hash)
+			// TODO: Make hash unique so we can ON CONFLICT REPLACE ALL here
```

Comment:
As a rule we don't really use TODO: in the code, prefer it to go into pivotal tracker. If you'd like to leave a TODO here we should link to the pivotal tracker story.

Reply:
I'll remove it, the story is here: https://www.pivotaltracker.com/story/show/172080031